compiler/rustc_mir_transform/src/remove_noop_landing_pads.rs

```diff
@@ -84,12 +83,12 @@ impl RemoveNoopLandingPads {
     fn remove_nop_landing_pads(&self, body: &mut Body<'_>) {
         debug!("body: {:#?}", body);
 
-        // make sure there's a resume block
-        let resume_block = {
-            let mut patch = MirPatch::new(body);
-            let resume_block = patch.resume_block();
-            patch.apply(body);
-            resume_block
+        // Find the resume block
+        let Some(resume_block) = body.basic_blocks.iter_enumerated().find(|(_bb, block)| {
+            matches!(block.terminator().kind, TerminatorKind::Resume) && block.statements.is_empty()
```

Comment:
Could we ever have a `resume` terminator with statements?
- if yes, we may need to create an empty resume block for this pass;
- if no, the `block.statements.is_empty()` would be better as an assertion + comment.

Reply:
I think the answer is yes, and the empty resume block is indeed needed for this pass. I'll need to use a different way to skip the empty resume block creation for nounwind functions.